### PR TITLE
Add recipe for dtaidistance

### PR DESCRIPTION
**Pull request by m-rossi**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/dtaidistance/meta.yaml
Comment (dopplershift) [inline recipes/dtaidistance/meta.yaml:28]: `cython` doesn't seem to be used at runtime:
```suggestion
```

File: recipes/dtaidistance/meta.yaml
Comment (dopplershift) [inline recipes/dtaidistance/meta.yaml:29]: Since it doesn't seem like a qt-based GUI is needed, better to take the bot up on its suggestion and depend on matplotlib-base:
```suggestion
    - matplotlib-base
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
A couple minor tweaks to the dependencies and this is good to go.

File: recipes/dtaidistance/meta.yaml
Comment (m-rossi) [reply recipes/dtaidistance/meta.yaml:28]: I also did not have `cython` at first listed here. However `pip check` failed and it seems to be listed in the requirements of the package: https://github.com/wannesm/dtaidistance/blob/v2.3.2/setup.py#L340

What do you recommend? Remove `pip check` or leave `cython` in run requirements?

Comment (dopplershift) [reply recipes/dtaidistance/meta.yaml:28]: Personally, I'd prefer not to force a build tool (Cython) on all the users of the package. But I'll leave it up to you as recipe maintainer to way that against the benefits of `pip check`.

Comment (m-rossi) [reply recipes/dtaidistance/meta.yaml]: I am of the same opinion, thanks for your feedback. Will discuss with the maintainer of the source repo when I add a conda-forge badge.

**Review by dopplershift — APPROVED** [review]
Pending CI.

**Pull request merged**
